src/views/Gallery.vue

```diff
@@ -9,110 +9,103 @@
-        </ion-grid>
-    </ion-content>
     <ion-content>
+      <div class="wrap">
+        <ion-spinner v-if="loading" name="dots"></ion-spinner>
+        <ion-grid v-else>
+          <ion-row>
+            <ion-col
+              size="4"
+              size-lg="2"
+              size-md="3"
+              v-for="(picture, index) in filteredPictures"
+              :key="index"
+            >
+              <div class="square"></div>
+              <div :key="index" class="gallery-img-wrap">
+                <img
+                  data-cy="image"
+                  class="gallery-img"
+                  :src="picture.file?.url"
+                  @click="viewPicture(picture, index)"
+                />
+              </div>
+            </ion-col>
+          </ion-row>
+        </ion-grid>
+      </div>
+      <ion-infinite-scroll @ionInfinite="loadMore($event)">
+        <ion-infinite-scroll-content
+          loading-spinner="crescent"
+          :loading-text="$t('generic.state.loadingMore')"
+        >
+        </ion-infinite-scroll-content>
+      </ion-infinite-scroll>
     </ion-content>
   </ion-page>
 </template>
 
 <script lang="ts">
 import {
   IonContent, IonPage,modalController,IonSearchbar,
-  IonCol,IonRow,IonGrid,IonThumbnail
+  IonCol,IonRow,IonGrid,
+  IonSpinner,
+  IonInfiniteScroll,
+  IonInfiniteScrollContent,
 } from '@ionic/vue';
 import { defineComponent, computed } from 'vue';
 import { useStore } from '../stores/';
 import PictureView from '../components/picture-view.vue';
-import { Model } from '@/types/model';
-// import PictureViewModal from '../components/picture-view-modal.vue';
 import TeamFilterHeader from '../components/team-filter-header.vue';
 
 export default defineComponent({
   name: 'Gallery',
   data(){
-      const pictureArray: any[] = [];
       return{
           searchValue:'',
-          selectedTeam:0,
-          pictureArray,
-          teamValue:''
       }
   },
    setup() {
     const store = useStore();
-    const pictureArr: any[]=[];
-    // const elem = store.getters["news/latest"];
     return {
-      elem: computed(() => store.getters["news/latest"]),
+      loading: computed(() => store.getters["feed/loading"]),
+      latestPosts: computed(() => store.getters["feed/entries"]),
       store,
-      pictureArr
+      loadMore: (ev: CustomEvent) => {
+        store.dispatch("feed/loadMore").then(() => {
+          (ev.target as any).complete();
+        });
+      },
     }
   },
   computed:{
-      filterPicture(){
-        const pictureList: any[] =[];
-        this.elem.find((x: any)=>{
-        const item = this.store.getters.objectsMap[x];
-         item.objects.find((x: Model) =>{ if(x.className == "Picture"){
-           pictureList.push(item.objects[0])
-          }});
-         });
+    filteredPosts(): any[] {
+      const postsWithPictures = this.latestPosts.filter((post: any) =>
+        post.objects.some((object: any) => object.className === "Picture")
+      );
```

Comment:
as mentioned in the previous PR, we have a dedicated `gallery`-store now that should be used instead. it also has a handy `selectTeam` faciity and is straight out easier to use.

One other benefit it has: it shows you all pictures you have access to, including those send to you via DM... this code will always only show the ones posted in the feed.